### ISSUE TITLE
ArgumentError when datadog integration is enabled

### ISSUE THREAD

**Issue by prodjito** [zendesk/racecar#234]
Hello, thanks for creating this Kafka consumer framework and integrating it with Datadog. I created a simple consumer and didn't change the boilerplater starter code. I added `datadog_enabled: true` in the `common` stanza of `config/racecar.yml`. When I ran the consumer, it immediate crashed with `ArgumentError: wrong number of arguments (given 4, expected 2..3)`. Looks like dogstatsd is expecting fewer arguments for a histogram than the consumer is actually passing. Any clue as to what I did wrong here? Thanks.

**Comment by dasch** [comment]
Can you share the exact version of the Datadog and Racecar gems?

**Comment by prodjito** [comment]
I should be using the latest version of both. I see the following in my test app's Gemfile.

`gem "racecar", "~> 2.2"

gem "dogstatsd-ruby", "~> 4.8"`

**Comment by dasch** [comment]
What does your `Gemfile.lock` say?

**Comment by dasch** [comment]
Also, can you paste the entire stack trace?

**Comment by prodjito** [comment]
Here is a partial list of what I see in my Gemfile.lock file under the `spec` stanza:
dogstatsd-ruby (4.8.3)
racecar (2.2.0)
  king_konf (\~> 1.0.0)
  rdkafka (\~> 0.8.0)

Here is the entire stack trace:
```=> Crashed: ArgumentError: wrong number of arguments (given 4, expected 2..3)
--- Caused by: ---
ArgumentError: wrong number of arguments (given 4, expected 2..3)
/root/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/dogstatsd-ruby-4.8.3/lib/datadog/statsd.rb:194:in `histogram'
/root/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/racecar-2.2.0/lib/racecar/datadog.rb:86:in `emit'
/root/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/racecar-2.2.0/lib/racecar/datadog.rb:79:in `block (2 levels) in <class:StatsdSubscriber>'
/root/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/racecar-2.2.0/lib/racecar/datadog.rb:175:in `main_loop'
/root/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/activesupport-6.1.3/lib/active_support/subscriber.rb:150:in `finish'
/root/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/activesupport-6.1.3/lib/active_support/notifications/fanout.rb:157:in `finish'
/root/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/activesupport-6.1.3/lib/active_support/notifications/fanout.rb:63:in `block in finish'
/root/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/activesupport-6.1.3/lib/active_support/notifications/fanout.rb:63:in `each'
/root/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/activesupport-6.1.3/lib/active_support/notifications/fanout.rb:63:in `finish'
/root/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/activesupport-6.1.3/lib/active_support/notifications/instrumenter.rb:45:in `finish_with_state'
/root/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/activesupport-6.1.3/lib/active_support/notifications/instrumenter.rb:30:in `ensure in instrument'
/root/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/activesupport-6.1.3/lib/active_support/notifications/instrumenter.rb:30:in `instrument'
/root/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/activesupport-6.1.3/lib/active_support/notifications.rb:203:in `instrument'
/root/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/racecar-2.2.0/lib/racecar/instrumenter.rb:25:in `instrument'
/root/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/racecar-2.2.0/lib/racecar/runner.rb:67:in `block in run'
/root/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/racecar-2.2.0/lib/racecar/runner.rb:64:in `loop'
/root/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/racecar-2.2.0/lib/racecar/runner.rb:64:in `run'
/root/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/racecar-2.2.0/lib/racecar.rb:54:in `run'
/root/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/racecar-2.2.0/lib/racecar/cli.rb:63:in `run'
/root/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/racecar-2.2.0/lib/racecar/cli.rb:13:in `main'
/root/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/racecar-2.2.0/exe/racecar:9:in `start'
/root/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/racecar-2.2.0/exe/racecar:44:in `<top (required)>'
/root/.rbenv/versions/3.0.0/bin/racecar:23:in `load'
/root/.rbenv/versions/3.0.0/bin/racecar:23:in `<main>'
```

**Comment by dasch** [comment]
Looks like dogstatsd-ruby made a backwards incompatible API change. Can you either figure out a way to support both APIs and do a PR on Racecar or, if you need to resolve this quickly, roll back to an earlier version of that gem?

**Comment by prodjito** [comment]
I tried several different versions of `dogstatsd-ruby` from 4.8.3 going back to 2.0.0, the same error occurs. Any idea what was the last version of `dogstatsd-ruby` that racecar is known to be compatible with?

I'm not a Ruby developer but from looking at `racecar.datadog.rb`, it seems every time `histogram` is invoked, 3 arguments are passed as parameters. Here is an example https://github.com/zendesk/racecar/blob/aab7dbd584c96b9167a16ad8444df5410285ef84/lib/racecar/datadog.rb#L126

I'm not sure why the error message is complaining that racecar is trying to pass 4 arguments.

**Comment by tjwp** [comment]
@prodjito from the stack trace I noticed that you're using Ruby 3.0.0. Could this be a difference with the argument handling in Ruby 3.0.0?

Are you able to test this with an earlier Ruby version?

**Comment by dasch** [comment]
Yeah, looks like Racecar doesn't work with Ruby 3's new parameter syntax: #237

**Comment by dasch** [comment]
Should be fixed by #237.
